### PR TITLE
Visualising changes in sentiment over time for a particular topic

### PR DESCRIPTION
**Pull request by ayahusseini**
- added slider for averaging interval
- added dropbox for topic selection

### CODE REVIEW

File: dashboard/visualise_time_changes.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+def resample_dataframe(df: pd.DataFrame, time_interval:str):
+    """Resamples the dataframe to return the average sentiment scores by source, topic 
+    over a set of grouped time intervals"""
+
+    df['date_published'] =pd.to_datetime(df['date_published'])
+
+    df_avg = df.groupby(['source_name', 'topic_name']).resample(time_interval, on= 'date_published').mean().reset_index()
+    return df_avg
+
+def generate_warning_message(source_to_topics: dict) -> str:
```
Comment (indiah444) [inline dashboard/visualise_time_changes.py:19]: Great idea to generating a warning message, will make the dashboard a lot easier to understand.

File: dashboard/db_functions.py [hunk 2]
```diff
@@ -37,3 +37,20 @@ def get_topic_dict() -> dict:
...
             res = cur.fetchall()
     return {topic['topic_name']: topic['topic_id'] for topic in res}
+
+def get_scores_topic(topic_name: str) -> dict:
+    """Returns a dictionary containing the polarity scores for a given topic """
+
+    topic_name = topic_name.strip().title()
+    with create_connection() as conn:
+        select_data = """SELECT t.topic_name, s.source_name, a.content_polarity_score, a.title_polarity_score, a.date_published FROM article a
+        INNER JOIN article_topic_assignment ata ON a.article_id = ata.article_id
```
Comment (YakMan101) [inline dashboard/db_functions.py:47]: Good consideration of JOIN type needed

File: dashboard/visualise_time_changes.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+"""Script to create visualisations of changes over time"""
+
+from datetime import datetime
```
Comment (YakMan101) [inline dashboard/visualise_time_changes.py:3]: Single line to separate native libraries vs external ones

File: dashboard/visualise_time_changes.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import streamlit as st
+import pandas as pd
+import altair as alt
+from db_functions import get_scores_topic, get_topic_names
```
Comment (YakMan101) [inline dashboard/visualise_time_changes.py:7]: Again single line to separate external libraries vs local files

File: dashboard/visualise_time_changes.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+
+def resample_dataframe(df: pd.DataFrame, time_interval:str):
+    """Resamples the dataframe to return the average sentiment scores by source, topic
```
Comment (YakMan101) [inline dashboard/visualise_time_changes.py:11]: Good docstring descriptions

**Comment by ayahusseini** [comment]
Need to check the merge (Conflict was that I added a function to db_functions"

**Review by megansam5 — APPROVED** [review]
Looks very nice! Will need to resolve merge conflicts in the database functions folder but then good to go!

**Pull request merged**